### PR TITLE
Update task editor to use Angular reactive forms

### LINKED ISSUE #1198: Task editor allows user to click "continue" when no questions/instructions are specified
1. Create survey
2. Add a question or other task without specifying a label
3. Click "Continue"

Expected:

* An error is shown and the user can't proceed to the next step

Actual:

* The creation flow proceeds to the next step.

@rfontanarosa PTAL? @os-micmec FYI

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1198

Angular Reactive Forms -> https://angular.io/guide/reactive-forms

Instead of using this.tasks and manage that list we can define a multi-level reactive form.

```
this.formGroup = new FormBuilder().group({
  id: this.dataStoreService.generateId(),
  type: types?.first(),
  required: false,
  label: ['', Validators.required],
  cardinality: null,
  options: new FormBuilder().array([]),
});
```
```
const formGroup = new FormBuilder().group({
  id: this.dataStoreService.generateId(),
  label: '',
  code: '',
});
```

https://github.com/google/ground-platform/assets/10047945/f13e9e02-48f1-48af-ad69-69e3b747adb2

### CODE REVIEW

**Comment by gino-m** [comment]
@rfontanarosa Please mark "Ready for review" once ready.

File: web/src/app/components/tasks-editor/tasks-editor.component.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  constructor(
+    private dataStoreService: DataStoreService,
+    private dialogService: DialogService
+  ) {}
+
+  ngOnChanges(): void {
+    const formBuilder = new FormBuilder();
+
+    if (this.tasks) {
+      this.formGroup = formBuilder.group({
```
Comment (gino-m) [inline web/src/app/components/tasks-editor/tasks-editor.component.ts:95]: This block looks intimidating :) Can we refactor it a bit into local vars and handler functions?

File: web/src/app/components/tasks-editor/tasks-editor.component.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      )
+      .afterClosed()
+      .subscribe(dialogResult => {
+        if (dialogResult) {
+          this.formArray.removeAt(index);
+        }
+      });
+  }
+
+  onDuplicateTask(index: number) {
```
Comment (gino-m) [inline web/src/app/components/tasks-editor/tasks-editor.component.ts:167]: I know this is still DRAFT - reminder to please add basic JSDoc comments to public methods.

File: web/src/app/components/tasks-editor/tasks-editor.component.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    this.dialogService
+      .openConfirmationDialog(
+        'Duplicate task',
+        'Are you sure you wish to duplicate this task?'
+      )
+      .afterClosed()
+      .subscribe(dialogResult => {
```
Comment (gino-m) [inline web/src/app/components/tasks-editor/tasks-editor.component.ts:174]: This lambda could also be broken out into its own function.

**Comment by rfontanarosa** [comment]
replaced by https://github.com/google/ground-platform/pull/1289

**Pull request closed without merging**